#! /bin/sh /usr/share/dpatch/dpatch-run
## 217-abend-on-mathlib-test-failure.dpatch by Kevin B. McCarty <kmccarty@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Force mathlib test program to exit non-zero if any test fails.

@DPATCH@
diff -urNad cernlib-2006.dfsg.2~/src/mathlib/gen/tests/main.F cernlib-2006.dfsg.2/src/mathlib/gen/tests/main.F
--- cernlib-2006.dfsg.2~/src/mathlib/gen/tests/main.F	1996-09-17 01:53:22.000000000 -0700
+++ cernlib-2006.dfsg.2/src/mathlib/gen/tests/main.F	2008-02-22 12:24:43.000000000 -0800
@@ -332,5 +332,7 @@
 #endif
 C     Close output streams
       IF (LOUT .NE. 6) CLOSE(UNIT=LOUT)
+C     Exit non-zero if any test failed           
+      IF (NFAIL .NE. 0) CALL EXITF(1)
       STOP
       END
